src/hooks/useSettings.ts

```diff
@@ -100,3 +100,61 @@ export const useFeatureEnabled = (featureName: FeatureSettingKey, roomId: string
 
     return enabled;
 };
+
+/**
+ * Retrieve a setting value, and listen for changes. This hook
+ * also returns a function that can update the setting with a new value.
+ * @param settingName
+ * @param level
+ * @param roomId
+ * @param excludeDefault
+ */
+export function useSettingsValueWithSetter<S extends SettingKey>(
+    settingName: S,
+    level: SettingLevel,
+    roomId: string | null,
+    excludeDefault: true,
+): [Settings[S]["default"] | undefined, (value: Settings[S]["default"]) => Promise<void>];
+export function useSettingsValueWithSetter<S extends SettingKey>(
+    settingName: S,
+    level: SettingLevel,
+    roomId?: string | null,
+    excludeDefault?: false,
+): [Settings[S]["default"], (value: Settings[S]["default"]) => Promise<void>];
+export function useSettingsValueWithSetter<S extends SettingKey>(
```

Comment:
Why do we need these functions?

```ts
const value = useSettingsValue("feature");
...
SettingStore.setValue("feature", newValue);
```
Is not enough?

Reply:
Ah, thinking about this again this makes a lot more sense. I've reverted these exports and done as you said.